google/cloud/spanner/bytes.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_BYTES_H_
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_BYTES_H_
+
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status_or.h"
+#include <array>
+#include <cstddef>
+#include <iterator>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+class Bytes;  // defined below
+
+// Internal forward declarations to befriend.
+namespace internal {
+StatusOr<Bytes> BytesFromBase64(std::string input);
+std::string BytesToBase64(Bytes const& b);
+}  // namespace internal
+
+/**
+ * A representation of the Spanner BYTES type: variable-length binary data.
+ *
+ * A `Bytes` value can be constructed from, and converted to any sequence of
+ * octets. All functions/operators should interpret the data as raw bytes (in
+ * contrast to the Spanner STRING type, where the data should be interpreted
+ * as Unicode characters).
```

Comment:
NO CHANGE NEEDED (feel free to commit as-is):

FWIW, I'm not sure if this last sentence is needed. It's talking about how the data should be interpreted. But `spanner::Bytes` is pretty much an opaque container, so callers can only even see the bytes either before they add them to `Bytes` or after they extract them from `Bytes`, at which point they can interpret the bytes however they want (they're their bytes).

Related: I'm not sure if mentioning Spanner STRING here clarifies or confuses anything. It's not clear why we are drawing any comparison between bytes and strings at all.

But just a question. Leaving as-is is fine with me. Whatever you think SGTM.